src/overlays/actors/ovl_En_Col_Man/z_en_col_man.c

```diff
@@ -39,34 +61,220 @@ const ActorInit En_Col_Man_InitVars = {
+        this->actor.shape.yOffset = 300.0f;
+        this->actor.shape.shadowScale = 5.0f;
+        if (this->actor.params == EN_COL_MAN_HEART_PIECE) {
+            this->actor.gravity = -2.0f;
+        }
+        this->unk148 = 1;
+    }
+    this->actionFunc = func_80AFDE00;
+}
+
+void func_80AFDE00(EnColMan* this, GlobalContext* globalCtx) {
+    if (this->actor.bgCheckFlags & 1) {
+        if (this->actor.params == EN_COL_MAN_HEART_PIECE) {
+            this->actor.params = EN_COL_MAN_RECOVERY_HEART;
+            this->actor.speedXZ = 2.0f;
+            this->actor.velocity.y = 8.0f;
+        } else {
+            this->actor.speedXZ = 0.0f;
+        }
+    }
+    if (!(gSaveContext.weekEventReg[0x38] & 2)) {
+        this->actor.shape.rot.y += 0x3E8;
+    }
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        this->actor.parent = NULL;
+        this->actor.draw = NULL;
+        this->actionFunc = EnColMan_SetHeartPieceCollectedAndKill;
+    } else if (!(gSaveContext.weekEventReg[0x38] & 2)) {
+        func_800B8A1C(&this->actor, globalCtx, GI_HEART_PIECE, 40.0f, 40.0f);
+    } else {
+        func_800B8A1C(&this->actor, globalCtx, GI_RECOVERY_HEART, 40.0f, 40.0f);
+    }
+}
+
+void EnColMan_SetHeartPieceCollectedAndKill(EnColMan* this, GlobalContext* globalCtx) {
+    if (func_80152498(&globalCtx->msgCtx) == 6 && func_80147624(globalCtx)) {
+        gSaveContext.weekEventReg[0x38] |= 2;
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void func_80AFDF60(EnColMan* this) {
+    this->actor.draw = func_80AFE584;
+    this->actor.flags |= 0x10;
+    this->actor.flags |= 0x20;
+    this->unk148 = 2;
+    this->actionFunc = func_80AFDFB4;
+    this->actor.shape.shadowScale = 5.0f;
+    this->actor.gravity = -3.0f;
+}
+
+void func_80AFDFB4(EnColMan* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f velocity;
+    Vec3f accel;
+
+    this->scale = (BREG(55) / 10000.0f) + 0.0015f;
+
+    if ((this->actor.bgCheckFlags & 1) && (this->actor.velocity.y < 0.0f)) {
+        if (!this->hasSetRandomValues) {
+            this->actor.world.rot.y = randPlusMinusPoint5Scaled(30000.0f);
+            this->actor.speedXZ = 2.0f + BREG(56) + Rand_ZeroFloat(2.0f);
+            this->actor.velocity.y = 12.0f + BREG(57) + Rand_ZeroFloat(5.0f);
+            this->hasSetRandomValues = true;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_ANSATSUSYA_ROCK);
+            return;
+        }
+
+        for (i = 0; i < 2; i++) {
+            velocity.x = randPlusMinusPoint5Scaled(2.0f);
+            velocity.y = Rand_ZeroFloat(2.0f) + 1.0f;
+            velocity.z = randPlusMinusPoint5Scaled(2.0f);
+            accel.y = -0.1f;
+            accel.z = 0.0f;
+            accel.x = 0.0f;
 
-extern ColliderCylinderInit D_80AFE730;
+            func_800B0EB0(globalCtx, &this->actor.world.pos, &velocity, &accel, &primColor, &envColor,
+                          Rand_ZeroFloat(50.0f) + 60.0f, 0x1E, Rand_ZeroFloat(5.0f) + 20.0f);
```

Comment:
The 0x1E is almost certainly actually 30, although it's not clear precisely what this function does. Timer or params for something would be my guess.